packages/arb-token-bridge-ui/src/components/TransactionHistory/PendingTransactions.tsx

```diff
@@ -64,10 +66,7 @@ export const PendingTransactions = ({
           <ExternalLink
             className="arb-hover cursor-pointer text-sm text-white underline"
             onClick={() => {
-              switchChain({
-                chainId: getOtherL2NetworkChainId(l2Network.chainID),
-                provider: provider as Web3Provider
-              })
+              switchNetwork?.(getOtherL2NetworkChainId(l2Network.chainID))
```

Comment:
General comment (applies to all instances):

When the wallet doesn't support automatic network switching, `switchNetwork` will be `undefined` ([source](https://wagmi.sh/react/hooks/useSwitchNetwork#usage)). In the previous implementation we had an alert pop up in that case, so we should do the same here. Maybe a custom `useSwitchNetwork` or `useSafeSwitchNetwork` that alerts `Please connect to ${network} in your wallet to continue.`.

Reply:
Will try this flag
https://wagmi.sh/react/hooks/useSwitchNetwork#throwforswitchchainnotsupported-optional

Reply:
catching it in `handleSwitchNetworkError`